bin/citrea/tests/test_client/mod.rs

```diff
@@ -691,6 +693,50 @@ impl TestClient {
         traces.into_iter().flatten().collect()
     }
 
+    pub(crate) async fn subscribe_new_heads(&self) -> mpsc::Receiver<RichBlock> {
+        let (tx, rx) = mpsc::channel();
+        let mut subscription = self
+            .ws_client
+            .subscribe("eth_subscribe", rpc_params!["newHeads"], "eth_unsubscribe")
+            .await
+            .unwrap();
+
+        tokio::spawn(async move {
```

Comment:
We have to find a way to decrease the number of `tokio::spawn` usages in our codebase and replace it somehow with workers we know are running. Point being, it's harder to debug randomly spawned tasks all over the codebase.